sdk/program/src/message/legacy.rs

```diff
@@ -44,6 +44,24 @@ lazy_static! {
     };
 }
 
+lazy_static! {
+    // A simple table to do a lookup based on the first byte of the static keys in question.
+    // Used as a very quick culling test before searching through vectors of keys for a match.
+    pub static ref MAYBE_BUILTIN_KEY_OR_SYSVAR: [bool; 256] = {
+        let mut temp_table: [bool; 256] = [false; 256];
+        BUILTIN_PROGRAMS_KEYS.iter().for_each(|key| temp_table[key.0[0] as usize] = true);
+        sysvar::ALL_IDS.iter().for_each(|key| temp_table[key.0[0] as usize] = true);
+        temp_table
+    };
```

Comment:
```suggestion
    const NUM_U8_INTS: usize = u8::MAX as usize + 1];
    pub static ref MAYBE_BUILTIN_KEY_OR_SYSVAR: [bool; NUM_U8_INTS] = {
        let mut temp_table: [bool; NUM_U8_INTS] = [false; NUM_U8_INTS];
        BUILTIN_PROGRAMS_KEYS.iter().for_each(|key| temp_table[key.0[0] as usize] = true);
        sysvar::ALL_IDS.iter().for_each(|key| temp_table[key.0[0] as usize] = true);
        temp_table
    };
```
My intent here is to make it more obvious why `MAYBE_BUILTIN_KEY_OR_SYSVAR[key.0[0] as usize]` won't ever panic on index out of bounds. I'm not in love with how this looks, though; can you think of a better way to make that explicit? ...Maybe a comment below would be enough.

Reply:
Hmm, yeah, I'm not sure that a const that's "how many different things fit into a u8" is making that clearer...  I can definitely add more comments, though.

Reply:
I mean, it's not even obvious without looking up the definition of Pubkey what `key.0[0]` is. Sure, the comment approach is fine.

Reply:
How about
// Each element of a key is a u8. We use key[0] as an index into this table of 256 boolean elements, to store
// whether or not the first element of any key is present in the static lists of built-in-program keys
// or all system ids. By using this lookup table, we can very quickly determine that a key under consideration
// cannot be in either of these lists.

Reply:
> I mean, it's not even obvious without looking up the definition of Pubkey what `key.0[0]` is. Sure, the comment approach is fine.
Took me a while to figure that out.

Reply:
Like it